valkey/templates/valkey-server/statefulset.yaml

```diff
@@ -0,0 +1,125 @@
+{{- $required_sentinels := int .Values.sentinel.replicas }}
+{{- range $_index, $server := .Values.valkey.servers }}
+{{- if eq $server.name "sentinel" }}
+{{- fail "The value 'sentinel' is not allowed for `.Values.valkey.servers.name`, it is reserved for the sentinel service" }}
+{{- end }}
+{{- $service_name := printf "valkey-server-%s" $server.name }}
+---
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: valkey-{{ $server.name }}
+  namespace: {{ $.Values.global.namespace }}
+  labels:
+    app.kubernetes.io/name: valkey-{{ $server.name }}
+    app.kubernetes.io/component: valkey-server
+    app.kubernetes.io/part-of: valkey
+spec:
+  serviceName: {{ $service_name }}
+  replicas: {{ default 2 $server.replicas | int }}
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: valkey-{{ $server.name }}
+      app.kubernetes.io/component: valkey-server
+      app.kubernetes.io/part-of: valkey
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: valkey-{{ $server.name }}
+        app.kubernetes.io/component: valkey-server
+        app.kubernetes.io/part-of: valkey
+    spec:
+      securityContext:
+        fsGroup: 1000
+
+      {{- if and (hasKey $.Values.global "topology_spread_constraints") ($.Values.global.topology_spread_constraints.enabled) }}
+      topologySpreadConstraints:
+        - labelSelector:
+            matchLabels:
+              app.kubernetes.io/name: valkey-{{ $server.name }}
```

Comment:
suggest we leave `name` as static and add `instance` with this value. And in sentinel, `name` is `sentinel` or `valkey-sentinel`, and then we don't need the guards at the top of the file.

Also the selector here should match the selector on line 20 ideally.